pkg/virtual/syncer/builder/build.go

```diff
@@ -134,57 +148,67 @@ func BuildVirtualWorkspace(
 			}
 
 			completedContext = genericapirequest.WithCluster(requestContext, cluster)
-			completedContext = syncercontext.WithSyncTargetName(completedContext, workloadCusterName)
+			completedContext = syncercontext.WithSyncTargetKey(completedContext, workloadv1alpha1.ToSyncTargetKey(logicalcluster.From(syncTargetObj), syncTargetObj.GetName()))
 			completedContext = dynamiccontext.WithAPIDomainKey(completedContext, apiDomainKey)
 			prefixToStrip = strings.TrimSuffix(urlPath, realPath)
 			accepted = true
 			return
-		}),
-		Authorizer: authorizer.AuthorizerFunc(func(ctx context.Context, a authorizer.Attributes) (authorizer.Decision, string, error) {
-			syncTargetKey := dynamiccontext.APIDomainKeyFrom(ctx)
-			negotiationWorkspaceName, syncTargetName := client.SplitClusterAwareKey(string(syncTargetKey))
+		})
+	}
 
-			authz, err := delegated.NewDelegatedAuthorizer(negotiationWorkspaceName, kubeClusterClient)
-			if err != nil {
-				return authorizer.DecisionNoOpinion, "Error", err
-			}
-			SARAttributes := authorizer.AttributesRecord{
-				User:            a.GetUser(),
-				Verb:            "sync",
-				Name:            syncTargetName,
-				APIGroup:        workloadv1alpha1.SchemeGroupVersion.Group,
-				APIVersion:      workloadv1alpha1.SchemeGroupVersion.Version,
-				Resource:        "synctargets",
-				ResourceRequest: true,
-			}
-			return authz.Authorize(ctx, SARAttributes)
-		}),
-		ReadyChecker: framework.ReadyFunc(func() error {
-			select {
-			case <-readyCh:
-				return nil
-			default:
-				return errors.New("syncer virtual workspace controllers are not started")
+	// Setup the APIReconciler indexes to share between both virtualworkspaces.
+	if err := wildcardKcpInformers.Workload().V1alpha1().SyncTargets().Informer().AddIndexers(cache.Indexers{
+		apireconciler.IndexSyncTargetsByExport: apireconciler.IndexSyncTargetsByExports,
+	}); err != nil {
+		return nil
+	}
+
+	if err := wildcardKcpInformers.Apis().V1alpha1().APIExports().Informer().AddIndexers(cache.Indexers{
+		apireconciler.IndexAPIExportsByAPIResourceSchema: apireconciler.IndexAPIExportsByAPIResourceSchemas,
+	}); err != nil {
+		return nil
+	}
+
+	buildRequirements := func(resourceState workloadv1alpha1.ResourceState) buildRequirementsFunc {
```

Comment:
```suggestion
	requirementsBuilderProvider := func(resourceState workloadv1alpha1.ResourceState) buildRequirementsFunc {
```

Currently 2 distinct things have the `buildRequirements` name: the one defined here, and the argument of the `bootstrapManagementProvider`, and they have distinct types. That's quite confusing.